### PR TITLE
Add kv examples for dev-reorg

### PR DESCRIPTION
**Pull request by maria-robobug**
Supplements https://github.com/couchbase/docs-server/pull/2275 

Adding some basic examples that we can use in the `dev-reorg` server docs: https://issues.couchbase.com/browse/DOC-9205

### CODE REVIEW

**Review by osfameron — COMMENTED** [review]
does merging release/3.2 into the PR apply cleanly when we merge the PR?

(I always rebase, as it feels less messy, but obviously doing it in github interface is a win too if it's not too confusing...)

File: modules/hello-world/examples/KvBulkHelloWorld.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        System.out.println("CAS: " + result.cas());
+      }
+      // end::kv-bulk-remove[]
+    }
+  }
+}
+
+
+// tag::kv-bulk-class[]
+class JsonDocument {
```
Comment (osfameron) [inline modules/hello-world/examples/KvBulkHelloWorld.java:132]: Is it best practice to define a class to store the tuple of id/value?

(e.g. as opposed to just using a Map and iterating over key/values?
Hmm, it's Java, so I guess an explicit new class may well be idiomatic... just seemed a bit overkill)

File: modules/hello-world/examples/KvHelloWorldScoped.java [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    Bucket bucket = cluster.bucket("travel-sample");
+    Collection hotelCollection = bucket.scope("inventory").collection("hotel");
+
+    {
+      System.out.println("Example: [kv-insert]");
+      // tag::kv-insert[]
+      // Create the document object.
+      JsonObject geo = JsonObject.create()
+          .put("lat", 51.35785)
+          .put("lon", 0.55818).put("accuracy", "RANGE_INTERPOLATED");
```
Comment (osfameron) [inline modules/hello-world/examples/KvHelloWorldScoped.java]: final `.put` on a new line for consistency

File: modules/hello-world/examples/KvHelloWorldScoped.java [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+      // Print the result's CAS metadata to the console.
+      System.out.println("CAS:" + upsertResult.cas());
+      // end::kv-update-upsert[]
+    }
+
+    {
+      System.out.println("Example: [kv-update-subdoc]");
+      // tag::kv-update-subdoc[]
+      List<MutateInSpec> specs = new ArrayList<>();
+      specs.add(MutateInSpec.upsert("pets_ok", true));
```
Comment (osfameron) [inline modules/hello-world/examples/KvHelloWorldScoped.java]: I've seen (and used) `= Arrays.asList(MutateInSpec.upsert(...))` elsewhere instead.

File: modules/test/test-hello-world.bats [hunk 4]
```diff
@@ -19,3 +19,13 @@ load 'test/test_helper.bash'
     assert_success
     assert_output --partial "RESULT: bar"
 }
+
+@test "[hello-world] - KvHelloWorldScoped.java" {
```
Comment (osfameron) [inline modules/test/test-hello-world.bats:23]: ah, I reordered some test examples in Howtos yesterday, as I was expecting them to be in Alphabetical order.
I see here that's not the case 😄

**Review by osfameron — APPROVED** [review]
Looks good 👍

File: modules/hello-world/examples/KvBulkHelloWorld.java (hunk 1 above)
Comment (osfameron) [reply modules/hello-world/examples/KvBulkHelloWorld.java:132]: 💡 trying to implement similar thing and have seen the light...

Comment (programmatix) [reply modules/hello-world/examples/KvBulkHelloWorld.java:132]: Yes, in Java you'd usually create a class for this.  But Java 14 added Records, which can do this with minimal boilerplate, so I'd use those now.

File: modules/hello-world/examples/KvHelloWorldScoped.java (hunk 3 above)
Comment (maria-robobug) [reply modules/hello-world/examples/KvHelloWorldScoped.java]: Same here, I think I've forgotten to do it in these examples, 👍 I'll update.

File: modules/hello-world/examples/KvBulkHelloWorld.java (hunk 1 above)
Comment (maria-robobug) [reply modules/hello-world/examples/KvBulkHelloWorld.java:132]: Worth discussing in our docs sync meeting with PMs, for now will leave as is, but would be really nice to update this to use Records!

**Comment by maria-robobug** [comment]
Start of the year seems like a good time for me to fix that flaky `Search.java` test...ignore the failure it's intermittent.

**Pull request merged**